serializer.go

```diff
@@ -18,9 +18,9 @@ package ionhash
 import "github.com/amzn/ion-go/ion"
 
 type serializer interface {
-	scalar(ionValue interface{})
+	scalar(ionValue interface{}) error
 
-	stepIn(ionValue interface{})
+	stepIn(ionValue interface{}) error
 
 	stepOut()
```

Comment:
minor question: Can this give an error?

Reply:
My PR https://github.com/amzn/ion-hash-go/pull/13 adds the error return to stepOut()

Reply:
Ah thanks. This is getting a little mind bending--trying to keep track of diffs across related PRs. Could we resolve the git conflicts in PR #9, commit the existing pending PRs. It would be nice to take a step back and see what we have done.